test/lib/PlacesTestUtils.js

```diff
@@ -69,7 +118,28 @@ const PlacesTestUtils = Object.freeze({
         }
       );
     });
-    return (yield promise);
+
+    yield notifPromise;
+    yield insertPromise;
+
+    // poll every 10ms until history items appear on disk
+    yield waitUntil(function*() {
```

Comment:
I like the early return statement and the other statement types, but I think that a loop with a conditional increment is more legible (and efficient).